src/util.js

```diff
@@ -276,6 +276,51 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
+// for debug methods
+function hookTo(prototypeObject, methodName, extension) {
+    var existingMethodImpl = prototypeObject[methodName];
+    if (typeof existingMethodImpl === "function" &&
+        existingMethodImpl.extensions) {
+        var extensions = existingMethodImpl.extensions;
+        extensions[extension.toString()] = extension;
+        // Performance optimization: remove noops
+        for (var key in extensions) {
+            if (extensions.hasOwnProperty(key) &&
+                extensions[key].toString().replace(/\s/g, "") ===
+                "function(){}") {
```

Comment:
Yes, using function.toString() was not a good idea, also because of cases like "function() {[native code]}". Will replace with static counter.